### ISSUE TITLE
Private method overloads are not stripped with --no-check

### ISSUE THREAD

**Issue by lucacasonato** [denoland/deno#11550]
```shell
$ deno run https://raw.githubusercontent.com/denodrivers/postgres/5464b016275eab54368e21714147bb6b1c0e40f9/client.ts
Check https://raw.githubusercontent.com/denodrivers/postgres/5464b016275eab54368e21714147bb6b1c0e40f9/client.ts

$ deno run --no-check https://raw.githubusercontent.com/denodrivers/postgres/5464b016275eab54368e21714147bb6b1c0e40f9/client.ts
error: Uncaught SyntaxError: Unexpected token ';'
    _query: Query<ResultType.ARRAY>,
                                   ^
    at <anonymous> (https://raw.githubusercontent.com/denodrivers/postgres/5464b016275eab54368e21714147bb6b1c0e40f9/client.ts:68:36)
```

Compiles fine with tsc, but emits invalid code when using swc: https://raw.githubusercontent.com/denodrivers/postgres/5464b016275eab54368e21714147bb6b1c0e40f9/client.ts.

**Comment by nayeemrmn** [comment]
I bypassed the source mapping, issue is the two overload declarations at https://github.com/denodrivers/postgres/blob/5464b016275eab54368e21714147bb6b1c0e40f9/client.ts#L67-L77 are left in the emit as:
```js
export class QueryClient {
  // ...
  #executeQuery(_query);
  #executeQuery(_query);
  #executeQuery(query) {
      return this.#connection.query(query);
  }
  // ...
}
```
cc @kdy1

**Comment by dsherret** [comment]
I'll fix this one in swc.

**Comment by bartlomieju** [comment]
Fixed in #11626
